tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Search.cs

```diff
@@ -284,136 +285,156 @@ public SearchResponseVM GetPinnedData(SearchRequestVM searchRequestVM, string li
                                 Dictionary<string, DocumentData> userpinnedDocumentCollection =
                                     JsonConvert.DeserializeObject<Dictionary<string, DocumentData>>(userPinnedMatter);
                                 searchResponse.TotalRows = userpinnedDocumentCollection.Count;
-                                //searchResponse.DocumentDataList = userpinnedDocumentCollection.Values.Reverse();
-                                if (searchRequestVM.SearchObject.Sort.Direction == 0)
+
+                                if (searchRequestVM.SearchObject.Sort.SortAndFilterPinnedData == false)
```

Comment:
If SortAndFilterPinnedData == false just return the pinned results with out sorting and filtering else sort or filter the pinned data